subsys/net/l2/ieee802154/ieee802154_shell.c

```diff
@@ -99,7 +99,7 @@ static int cmd_ieee802154_associate(const struct shell *sh,
 
 	params = (struct ieee802154_req_params){0};
 	params.pan_id = atoi(argv[1]);
-	strncpy(ext_addr, argv[2], sizeof(ext_addr));
+	strncpy(ext_addr, argv[2], EXT_ADDR_STR_LEN);
```

Comment:
Why this change is needed, the sizeof() should have exactly the same value as the define?

Reply:
See https://github.com/zephyrproject-rtos/zephyr/blob/9f60075d14a283602a3580b6b97dd4686b53a28e/subsys/net/l2/ieee802154/ieee802154_shell.c#L25

The length of ext_addr is EXT_ADDR_STR_SIZE. 
EXT_ADDR_STR_LEN = EXT_ADDR_STR_SIZE - 1;

In line 103, we are using `strlen` which expects a NULL terminator. Thus we should only copy `sizeof(ext_addr) - 1` number of elements.

Reply:
Ah, my bad, I did not spot that there was two different defines, the other with SIZE and the other LEN. Sorry for the noise.